api/application_msgraph.go

```diff
@@ -15,21 +15,33 @@ import (
 	"github.com/hashicorp/go-multierror"
 )
 
-const (
-	// DefaultGraphMicrosoftComURI is the default URI used for the service MS Graph API
-	DefaultGraphMicrosoftComURI = "https://graph.microsoft.com"
-)
-
 var _ ApplicationsClient = (*AppClient)(nil)
 var _ GroupsClient = (*AppClient)(nil)
 var _ ServicePrincipalClient = (*AppClient)(nil)
 
 type AppClient struct {
-	client authorization.BaseClient
+	client   authorization.BaseClient
+	graphURI string
+}
+
+// Reference: https://docs.microsoft.com/en-us/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints
+func GetGraphURI(env string) (string, error) {
+	switch env {
+	case "AzurePublicCloud":
```

Comment:
I wonder if we should default empty string to public cloud instead of error, if that happens to be the case for whatever reason.
```suggestion
	case "AzurePublicCloud", "":
```